### PR TITLE
Rust embedded hal 1 0

### PR DESCRIPTION
**Pull request by engdoreis**
Update the ibex-demo-system embedded-hal implementation to the release version 1.0.0

### CODE REVIEW

File: sw/rust/demo/lcd_hal/src/main.rs [hunk 1]
```diff
@@ -23,8 +24,7 @@ use st7735_lcd::Orientation;
 
 use crate::hal::{pac, GpioExt};
 
-use embedded_hal;
-use embedded_hal::{blocking::delay::DelayMs, digital::v2::OutputPin};
+use embedded_hal::{self, delay::DelayNs, digital::OutputPin};
```
Comment (marnovandermaas) [inline sw/rust/demo/lcd_hal/src/main.rs:27]: Does the `DelayNs` package still have the `delay_ms` function?

File: sw/rust/demo/led_hal/src/main.rs [hunk 2]
```diff
@@ -78,6 +78,6 @@ fn main() -> ! {
                 }
             }
         }
-        delay.delay_ms(50u16);
+        delay.delay_ms(50);
```
Comment (marnovandermaas) [inline sw/rust/demo/led_hal/src/main.rs:81]: Why not `50u32` like you did above?

File: sw/common/link.ld [hunk 3]
```diff
@@ -41,6 +41,22 @@ PROVIDE(_mp_hook = default_mp_hook);
...
+PROVIDE(LoadFault = ExceptionHandler);
+PROVIDE(StoreMisaligned = ExceptionHandler);
+PROVIDE(StoreFault = ExceptionHandler);
+PROVIDE(UserEnvCall = ExceptionHandler);
+PROVIDE(SupervisorEnvCall = ExceptionHandler);
+PROVIDE(MachineEnvCall = ExceptionHandler);
+PROVIDE(InstructionPageFault = ExceptionHandler);
+PROVIDE(LoadPageFault = ExceptionHandler);
+PROVIDE(StorePageFault = ExceptionHandler);
+
```
Comment (marnovandermaas) [inline sw/common/link.ld:59]: Can we only change the link file for rust programs and not for C as well.

File: sw/rust/demo/lcd_hal/src/main.rs (hunk 1 above)
Comment (engdoreis) [reply sw/rust/demo/lcd_hal/src/main.rs:27]: Yes.
https://docs.rs/embedded-hal/1.0.0/embedded_hal/delay/trait.DelayNs.html

File: sw/common/link.ld (hunk 3 above)
Comment (engdoreis) [reply sw/common/link.ld:59]: I think that it doesn't hurt to have these symbols on a common linker script, the C Firmware just won't use them.

**Pull request merged**